### PR TITLE
Add Tool to inspect DDS header and preview texture

### PR DESCRIPTION
**Pull request by lemirep**
Task-Id: KUE-515

### CODE REVIEW

File: tools/ddspreviewer/TexturePreview.qml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    Camera {
+        id: camera
+        fieldOfView: 40
+        position: Qt.vector3d(0, 0, -5)
+        upVector: Qt.vector3d(0, 1, 0)
+        viewCenter: Qt.vector3d(0, 0, 0)
+    }
+
+    OrbitCameraController {
```
Comment (er-vin) [inline tools/ddspreviewer/TexturePreview.qml]: Shouldn't this be disabled in plane mode? Or there's a use I don't see of moving the camera around in that mode?

File: tools/ddspreviewer/TexturePreview.qml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    KUE.Skybox {
+        enabled: mode === _SKYBOX_MODE
+        extension: ".dds"
+        readonly property int ddsExtIdx: filePath.lastIndexOf(".dds")
+        baseName: ddsExtIdx !== -1
+                  ? filePath.substring(0, ddsExtIdx)
+                  : filePath
+        onBaseNameChanged: console.log("BaseName " + baseName)
```
Comment (er-vin) [inline tools/ddspreviewer/TexturePreview.qml]: Are we sure we want to keep that debug?

File: tools/ddspreviewer/ddsheaderparser.h [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    ~DDSHeaderParser();
+
+    QUrl filePath() const;
+    QString ddsHeader() const;
+
+public slots:
+    void setFilePath(QUrl filePath);
+
+signals:
+    void filePathChanged(QUrl filePath);
```
[the remark this answers is not in the document]
Comment (lemirep) [reply tools/ddspreviewer/ddsheaderparser.h]: Done

File: tools/ddspreviewer/ddsheaderparser.h [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    Q_ENUM(DXGIFormat)
+
+    explicit DDSHeaderParser(QObject *parent = nullptr);
+    ~DDSHeaderParser();
+
+    QUrl filePath() const;
+    QString ddsHeader() const;
+
+public slots:
+    void setFilePath(QUrl filePath);
```
[the remark this answers is not in the document]
Comment (lemirep) [reply tools/ddspreviewer/ddsheaderparser.h]: Done

File: tools/ddspreviewer/ddsheaderparser.cpp [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+void DDSHeaderParser::parseTextureHeader()
+{
+    m_ddsHeader.clear();
+    emit ddsHeaderChanged();
+
+    // 1) Check path and that file exists
+    QFileInfo fInfo(m_filePath.path());
+    // 2) Check extension is .dds
+    qDebug() << Q_FUNC_INFO << fInfo.exists() << fInfo.suffix();
```
[the remark this answers is not in the document]
Comment (lemirep) [reply tools/ddspreviewer/ddsheaderparser.cpp]: Done

File: tools/ddspreviewer/ddsheaderparser.cpp [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+{
+    quint32 format;
+    quint32 dimension;
+    quint32 miscFlag;
+    quint32 arraySize;
+    quint32 miscFlags2;
+};
+
+template <int a, int b, int c, int d>
+struct DdsFourCC
```
[the remark this answers is not in the document]
Comment (lemirep) [reply tools/ddspreviewer/ddsheaderparser.cpp]: done

File: tools/ddspreviewer/TexturePreview.qml (hunk 1 above)
Comment (lemirep) [reply tools/ddspreviewer/TexturePreview.qml]: you probably still want to zoom in out in some cases

File: tools/ddspreviewer/TexturePreview.qml (hunk 2 above)
Comment (lemirep) [reply tools/ddspreviewer/TexturePreview.qml]: done

**Comment by lemirep** [comment]
I guess we can wait a bit indeed, we might not be too far from having something that works fully

File: tools/ddspreviewer/ddsheaderparser.cpp [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+        qWarning() << "Invalid DDS Header";
+        return;
+    }
+
+    const quint32 pixelFlags = qFromLittleEndian(header.pixelFormat.flags);
+    const quint32 fourCC = qFromLittleEndian(header.pixelFormat.fourCC);
+
+    QString dx10HeaderStr = QStringLiteral("{}");
+
+    if (fourCC == DdsFourCC('D', 'X', '1', '0')) {
```
Comment (er-vin) [inline tools/ddspreviewer/ddsheaderparser.cpp]: To be 100% sure this is computed by the compiler only once you probably want a constexpr intermediate:
constexpr DdsFourCC dx10FourCC{'D', 'X', '1', '0'};
if (fourCC == dx10FourCC) {

Maybe with C++20 we won't need this dance anymore...

**Pull request merged**